### PR TITLE
Added dnsapi/dns_infomaniak.sh

### PR DESCRIPTION
**Pull request by reneluria**
Added dns_infomaniak.sh to dnsapi to support infomaniak.com public API

report bugs to https://github.com/acmesh-official/acme.sh/issues/3188

### CODE REVIEW

**Comment by reneluria** [comment]
https://github.com/Infomaniak/acme.sh/actions/runs/265167704

File: dnsapi/dns_infomaniak.sh [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+  fqdn=${fulldomain#_acme-challenge.}
+
+  # guess which base domain to add record to
+  zone_and_id=$(_find_zone "$fqdn")
+  if [ -z "$zone_and_id" ]; then
+    _err "cannot find zone to modify"
+    return 1
+  fi
+  tmpfile=$(mktemp -u)
```
Comment (Neilpang) [inline dnsapi/dns_infomaniak.sh:149]: what is this tmpfile for ?

Comment (reneluria) [reply dnsapi/dns_infomaniak.sh:149]: it is a trick to read separated variables afterwards, but I can do it without the tmpfile

**Comment by reneluria** [comment]
What do you think of it now ?

**Comment by Neilpang** [comment]
It seems that  the dns test in docker has some failing cases.
please rebase to the lastest dev code, and run the tests again.

**Comment by reneluria** [comment]
Hello, I understand but the test fails if it cannot fetch the secret.
It can fetch the secret when running in my fork, but when run in the pull request, the Action doesn't have access to the secret in my repo, right ?

**Comment by Neilpang** [comment]
Yes,  I see.
I just opened the link to your Actions,  the result there was not passed either.

**Comment by reneluria** [comment]
But here it is:
https://github.com/Infomaniak/acme.sh/actions/runs/281862135

the previous runs failed on the OSX action due to instability at homebrew servers

File: dnsapi/dns_infomaniak.sh [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+{/g' | sed -n 's/.*"id":"*\([0-9]*\)"*.*"source_idn":"'"$fulldomain"'".*"target_idn":"'"$txtvalue"'".*/\1/p')
+  if [ -z "$record_id" ]; then
+    _err "could not find record to delete"
+    return 1
+  fi
+  _debug "record_id: $record_id"
+
+  # API call
+  response=$(_post "" "${INFOMANIAK_API_URL}/1/domain/$domain_id/dns/record/$record_id" "" DELETE)
+  if [ -n "$response" ] && echo "$response" | grep -qF '"result":"success"'; then
```
Comment (Neilpang) [inline dnsapi/dns_infomaniak.sh:168]: please don't use `grep -qF`
Use `_contains()` function instead.

File: dnsapi/dns_infomaniak.sh [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  # shellcheck disable=SC1004
+  _get "${INFOMANIAK_API_URL}/1/product?service_name=domain&customer_name=$domain" | sed 's/.*"data":\[{\(.*\)}\]}/\1/; s/,/\
+/g' | sed -n 's/^"id":\(.*\)/\1/p'
+}
+
+_find_zone() {
+  zone="$1"
+
+  # find domain in list, removing . parts sequentialy
+  while echo "$zone" | grep -q '\.'; do
```
Comment (Neilpang) [inline dnsapi/dns_infomaniak.sh:190]: don't use `grep -q`
please use pip ` >/dev/null` instead.

File: dnsapi/dns_infomaniak.sh (hunk 2 above)
Comment (reneluria) [reply dnsapi/dns_infomaniak.sh:168]: Ok, I'll do that

File: dnsapi/dns_infomaniak.sh (hunk 3 above)
Comment (reneluria) [reply dnsapi/dns_infomaniak.sh:190]: Same here, I'll remove all grep occurences

**Comment by Neilpang** [comment]
It seems that the centos:latest test is failed.
https://github.com/Infomaniak/acme.sh/runs/1198344430?check_suite_focus=true

Can you plese re-run the test jobs again?

**Comment by Neilpang** [comment]
I checked your Actions log. it seems  docker CentOS:latest is falied. centos:latest [FAIL]
please re-run your test jobs, I belive your will see the errors.

**Comment by reneluria** [comment]
@Neilpang yes now it's ok 
https://github.com/Infomaniak/acme.sh/actions/runs/297133086
only for the FreeBSD part but I see you are working on it

File: .github/workflows/DNS.yml [hunk 4]
```diff
@@ -75,8 +75,8 @@ jobs:
         if [ "${{ secrets.TokenName5}}" ] ; then
           echo "${{ secrets.TokenName5}}=${{ secrets.TokenValue5}}" >> env.list
         fi
-        echo "TEST_DNS_NO_WILDCARD" >> env.list 
-        echo "TEST_DNS_SLEEP" >> env.list
+        echo "TEST_DNS_NO_WILDCARD=$TEST_DNS_NO_WILDCARD" >> env.list
```
Comment (Neilpang) [inline .github/workflows/DNS.yml:78]: I don't think this change is necessary. 
please revert it

**Comment by Neilpang** [comment]
the freebsd is fixed.
please rebase code and run the test again.

File: .github/workflows/DNS.yml (hunk 4 above)
Comment (reneluria) [reply .github/workflows/DNS.yml]: Ok

**Comment by reneluria** [comment]
There you go:
https://github.com/Infomaniak/acme.sh/actions/runs/299370212

**Comment by easybe** [comment]
Thanks, perfect timing!

**Pull request merged**